### ISSUE TITLE
Fix retain_state=True

### ISSUE THREAD

**Issue by verbiiyo** [McCrearyD/PAGNNs#2]
backward pass raises flags because when we calculate the gradients w.r.t the PAGNN’s state we can’t do so a second time since the leaf node is removed from the compute graph. Either use loss.backward(retain_graph=True) (not great) or find a workaround. This either means we have to clip the compute graph (possibly a good thing) or we would have to retain the graph which could lead to vanishing gradients anyways.
